### DISCUSSION TITLE
Reading the context variables

### DISCUSSION THREAD

**Discussion by nepomuk16321** [EDCD/EDDI#2657] [Troubleshoot Scripts] [answered]
Question or what am I doing wrong?
Why does EDDI, when I use the context variables, always tell me the second last variable and not the last variable when I call it in another script? I have tried with the original context variables and with self-created variables. Example: In the script “_Body scanned_”, the name of the body is written to the variable “_eddi_context_body_name_” (read via _event.bodyname_) >>` {SetState('eddi_context_body_name', event.bodyname)}`. If I call the variable in another event script, I am told the name of the previously scanned body and not the name of the last scanned body.
Call with `{state.eddi_context_body_name}` e.g. in the script “_Surface signals detected_”. ...

**Comment by Tkael** [comment, marked as answer]
I think this is the same issue we looked at previously? https://github.com/EDCD/EDDI/issues/2647
The fix for this hasn't been released yet (hopefully soon!)
